src/components/common/CropModal.tsx

```diff
@@ -31,68 +54,83 @@ const CropModal = ({
   onClose,
   src,
 }: CropProps): JSX.Element => {
-  const [crop, setCrop] = useState<PixelCrop>();
-  const imageRef = useRef<HTMLImageElement>();
+  const [crop, setCrop] = useState<Crop>();
+  const [completedCrop, setCompletedCrop] = useState<PixelCrop>();
+  const imageRef = useRef<HTMLImageElement | null>(null);
   const { t } = useBuilderTranslation();
 
-  const makeClientCrop = async (newCrop: PixelCrop) => {
-    if (imageRef.current && newCrop.width && newCrop.height) {
-      const croppedImage = await getCroppedImg(imageRef.current, newCrop);
-      return croppedImage;
+  const handleOnConfirm = async () => {
+    const image = imageRef.current;
+    if (!image || !completedCrop) {
+      throw new Error('Crop canvas does not exist');
```

Comment:
@pyphilia Interesting, since it is inside an async function, it just sends a `uncaught in Promise` and the app does not crash ! Should we simply log an error and return ? How bad do we think it can be ?